app/controllers/download_controller.rb

```diff
@@ -39,18 +39,17 @@ def create
   def success; end
 
   def download_json
-    data = open("#{@register.url}/records.json?page-size=5000", &:read)
+    data = download_from_uri("#{@register.url}/records.json?page-size=5000")
     send_data data, type: "application/json; header=present", disposition: "attachment; filename=#{@register.slug}.json"
   end
 
   def download_csv
-    data = open("#{@register.url}/records.csv?page-size=5000", &:read)
+    data = download_from_uri("#{@register.url}/records.csv?page-size=5000")
```

Comment:
since these arguments are the same except the format, maybe we change the argument from the URL to the format, and build the URL in the shared method?
e.g. `download_with_format('csv')`?